### ISSUE TITLE
hardfault 

### ISSUE THREAD

**Issue by stmoon** [PX4/Firmware#8973]
Hello,

I found the hardfault with firmware px4fmu-v3_default (v1.7.3)

Could you tell why the hardfault happens and how to fix it?

I link the ulog (https://logs.px4.io/plot_app?log=71cff15b-4655-4f6b-a35e-e5675c057866)

Thanks in advance

```
[hardfault_log] -- 2000-01-01-01:46:31 Begin Fault Log --
System fault Occurred on: 2000-01-01-01:46:31
Type:Hard Fault in file:armv7-m/up_hardfault.c at line: 171 running task: mavlink_rcv_if0
FW git-hash: f293593a19605c0de63d24d2ef1456ce3349e8bc
Build datetime: Feb 19 2018 10:41:16
Build url: localhost
Processor registers: from 0x20014ec4
r0:0x100079a0 r1:0x200145a8 r2:0x10004c08 r3:0x20003ab0 r4:0x00000000 r5:0x08360000 r6:0x28f9004a r7:0x20015220
r8:0x0813c1cc r9:0x00000000 r10:0x0000001c r11:0x00000000 r12:0x000000c8 sp:0x20014f98 lr:0x0800b98f pc:0x000bdace
xpsr:0x00000000 basepri:0x000000f0 control:0x00000004
exe return:0xffffffe9
IRQ stack:
top: 0x200029f0
sp: 0x200029a0 Valid
bottom: 0x20002704
size: 0x000002ec
used: 000000c8
User stack:
top: 0x20015210
sp: 0x20014f98 Valid
bottom: 0x200146fc
size: 0x00000b14
used: 00000b14
Interrupt sp memory region, stack pointer lies within stack
0x20002a51 0x10006870
0x20002a50 0x000000b7
0x20002a4f 0x20010780
0x20002a4e 0x000000b6
0x20002a4d 0x10007470
0x20002a4c 0x00000075
0x20002a4b 0x1000e190
0x20002a4a 0x00000114
0x20002a49 0x20017520
0x20002a48 0x000000b3
0x20002a47 0x20007fa0
0x20002a46 0x00000112
0x20002a45 0x200176b0
0x20002a44 0x00000071
0x20002a43 0x1000c900
0x20002a42 0x000000d0
0x20002a41 0x20014530
0x20002a40 0x000000cf
0x20002a3f 0x1000e380
0x20002a3e 0x0000ffff
0x20002a3d 0x00000000
0x20002a3c 0x0000ffff
0x20002a3b 0x00000000
0x20002a3a 0x0000ffff
0x20002a39 0x00000000
0x20002a38 0x0000ffff
0x20002a37 0x00000000
0x20002a36 0x0000ffff
0x20002a35 0x00000000
0x20002a34 0x00000109
0x20002a33 0x20015f70
0x20002a32 0x0000ffff
0x20002a31 0x00000000
0x20002a30 0x000000c7
0x20002a2f 0x2000e430
0x20002a2e 0x0000ffff
0x20002a2d 0x00000000
0x20002a2c 0x0000ffff
0x20002a2b 0x00000000
0x20002a2a 0x0000ffff
0x20002a29 0x00000000
0x20002a28 0x00000003
0x20002a27 0x10002620
0x20002a26 0x00000002
0x20002a25 0x10001740
0x20002a24 0x00000001
0x20002a23 0x10000860
0x20002a22 0x00000000
0x20002a21 0x20002ccc
0x20002a20 0x20010780
0x20002a1f 0x2000e430
0x20002a1e 0x00000000
0x20002a1d 0x00000000
0x20002a1c 0x00000000
0x20002a1b 0x00000000
0x20002a1a 0x00000000
0x20002a19 0x00000000
0x20002a18 0x00000000
0x20002a17 0x00000000
0x20002a16 0x00000000
0x20002a15 0x20002b58
0x20002a14 0x00000000
0x20002a13 0x00000000
0x20002a12 0x00000000
0x20002a11 0x20002b48
0x20002a10 0x20002b68
0x20002a0f 0x00000000
0x20002a0e 0x00000000
0x20002a0d 0x20002b38
0x20002a0c 0x20002b58
0x20002a0b 0x00000000
0x20002a0a 0x00000000
0x20002a09 0x20002b28
0x20002a08 0x20002b48
0x20002a07 0x00000000
0x20002a06 0x00000000
0x20002a05 0x20002b18
0x20002a04 0x20002b38
0x20002a03 0x00000000
0x20002a02 0x00000000
0x20002a01 0x2001ebc8
0x20002a00 0x20002b28
0x200029ff 0x00000000
0x200029fe 0x00000000
0x200029fd 0x20002af8
0x200029fc 0x2001ebc8
0x200029fb 0x00000000
0x200029fa 0x00000000
0x200029f9 0x20002ae8
0x200029f8 0x20002b08
0x200029f7 0x00000000
0x200029f6 0x00000000
0x200029f5 0x20002ad8
0x200029f4 0x20002af8
0x200029f3 0x00000000
0x200029f2 0x00000000
0x200029f1 0x2001dab8
0x200029f0 0x20002ae8<-- Interrupt sp top
0x200029ef 0x00000000
0x200029ee 0x00000000
0x200029ed 0x20002ab8
0x200029ec 0x2001dab8
0x200029eb 0x00000000
0x200029ea 0x00000000
0x200029e9 0x20002aa8
0x200029e8 0x20002ac8
0x200029e7 0x00000000
0x200029e6 0x00000000
0x200029e5 0x20002a98
0x200029e4 0x20002ab8
0x200029e3 0x00000000
0x200029e2 0x00000000
0x200029e1 0x20002a88
0x200029e0 0x20002aa8
0x200029df 0x00000000
0x200029de 0x00000000
0x200029dd 0x1000e888
0x200029dc 0x20002a98
0x200029db 0x00000000
0x200029da 0x00000000
0x200029d9 0x20017f38
0x200029d8 0x1000e888
0x200029d7 0x00000000
0x200029d6 0x00000000
0x200029d5 0x20002a58
0x200029d4 0x20017f38
0x200029d3 0x00000000
0x200029d2 0x00000000
0x200029d1 0x100045e8
0x200029d0 0x20002a68
0x200029cf 0x00000000
0x200029ce 0x00000000
0x200029cd 0x00000000
0x200029cc 0x10007ca8
0x200029cb 0x00000000
0x200029ca 0x00000000
0x200029c9 0x00000002
0x200029c8 0x1000fff8
0x200029c7 0x2002fff8
0x200029c6 0x10000000
0x200029c5 0x20006ba0
0x200029c4 0x00039460
0x200029c3 0x00000000
0x200029c2 0x0000ffff
0x200029c1 0x00000000
0x200029c0 0x00000000
0x200029bf 0x00000000
0x200029be 0x00000000
0x200029bd 0x00000001
0x200029bc 0x10000010
0x200029bb 0x0000ffff
0x200029ba 0x00000000
0x200029b9 0x00000000
0x200029b8 0x00000000
0x200029b7 0x00000000
0x200029b6 0x00000001
0x200029b5 0x00000001
0x200029b4 0x20014ec4
0x200029b3 0x08004f33
0x200029b2 0x20015220
0x200029b1 0x28f9004a
0x200029b0 0x08360000
0x200029af 0x20014ec4
0x200029ae 0x000000f0
0x200029ad 0x080059bd
0x200029ac 0x080059c9
0x200029ab 0x080059eb
0x200029aa 0x00000000
0x200029a9 0x00000003
0x200029a8 0x080059e3
0x200029a7 0x200029c8
0x200029a6 0x000bdace
0x200029a5 0x0800b98f
0x200029a4 0x20014f98
0x200029a3 0x000000c8
0x200029a2 0x00000000
0x200029a1 0x00000000
0x200029a0 0x000000f0<-- Interrupt sp
0x2000299f 0x08005939
0x2000299e 0x000000ab
0x2000299d 0x08112b6b
0x2000299c 0x20002ca8
0x2000299b 0x200029f0
0x2000299a 0x00000000
0x20002999 0x200029f0
0x20002998 0x000000ab
0x20002997 0x08008f97
0x20002996 0x20000010
0x20002995 0x08005ac9
0x20002994 0x00000000
0x20002993 0x20014530
0x20002992 0x0800b98f
0x20002991 0x20014ec4
0x20002990 0x200029f0
0x2000298f 0x200029a0
0x2000298e 0x000000d0
0x2000298d 0x20014530
0x2000298c 0x20000000
0x2000298b 0x08005d49
0x2000298a 0x20002960
0x20002989 0x20002960
0x20002988 0x00000000
0x20002987 0x08005d2b
0x20002986 0x0000003f
0x20002985 0x10009270
0x20002984 0x00000000
0x20002983 0x00000037
0x20002982 0x08013749
0x20002981 0x080069b5
0x20002980 0x08013749
0x2000297f 0x0800696f
0x2000297e 0x20014530
0x2000297d 0x0800b98f
0x2000297c 0x000000ab
0x2000297b 0x20002ba4
0x2000297a 0x00000000
0x20002979 0x08005765
0x20002978 0x20002944
0x20002977 0x081129b0
0x20002976 0x00000000
0x20002975 0x00000000
0x20002974 0x2000295c
0x20002973 0x00000004
0x20002972 0x08013749
0x20002971 0x08013729
0x20002970 0x0000000a
0x2000296f 0x08004665
0x2000296e 0x08118b25
0x2000296d 0x00000039
0x2000296c 0x00000004
0x2000296b 0x080133cf
0x2000296a 0x20002924
0x20002969 0x00000020
0x20002968 0x080069b5
0x20002967 0x080069c1
0x20002966 0x20002924
0x20002965 0x0000000a
0x20002964 0x0000000a
0x20002963 0x08004da5
0x20002962 0x0000000a
0x20002961 0x08004665
0x20002960 0x00000010
0x2000295f 0x08004da5
0x2000295e 0x00000037
0x2000295d 0x08004665
0x2000295c 0x00000010
0x2000295b 0x08004da5
0x2000295a 0x00000038
0x20002959 0x08004665
0x20002958 0x00000010
0x20002957 0x08004da5
0x20002956 0x00000031
0x20002955 0x08004665
0x20002954 0x00000010
0x20002953 0x20002858
0x20002952 0x00000001
0x20002951 0x00000003
0x20002950 0xb5445bbe
0x2000294f 0x3671df7e
0x2000294e 0xb5fbde42
0x2000294d 0x20002840
0x2000294c 0x00000001
0x2000294b 0x00000003
0x2000294a 0xb63da751
0x20002949 0x373b7b0f
0x20002948 0xb77f8b46
0x20002947 0x08025a47
0x20002946 0x20008fdc
0x20002945 0x20008fb0
0x20002944 0x20008fb0
0x20002943 0x080de723
0x20002942 0x200028f8
0x20002941 0x2000a120
0x20002940 0x2000a350
0x2000293f 0x08005ac9
0x2000293e 0x00000000
0x2000293d 0x00000000
0x2000293c 0x00000000
0x2000293b 0x00000000
0x2000293a 0x200028e0
0x20002939 0x00000000
0x20002938 0x00000000
0x20002937 0x00000002
0x20002936 0x2000a330
0x20002935 0x2cb97c79
0x20002934 0xad469d29
0x20002933 0x200027d8
0x20002932 0x00000001
0x20002931 0x00000003
0x20002930 0x369c7815
0x2000292f 0x36cd9b9e
0x2000292e 0x36901638
0x2000292d 0x200027c0
0x2000292c 0x00000001
0x2000292b 0x00000003
0x2000292a 0x349a918e
0x20002929 0x350e3c42
0x20002928 0x34aeb6af
0x20002927 0x200027a8
0x20002926 0x00000001
0x20002925 0x00000003
0x20002924 0x35688154
0x20002923 0x366edb82
0x20002922 0x3601652d
0x20002921 0x08025837
0x20002920 0x00300aa8
0x2000291f 0x200028c8
0x2000291e 0x20009e78
0x2000291d 0xdeadbeef
0x2000291c 0xdeadbeef
0x2000291b 0xdeadbeef
0x2000291a 0xdeadbeef
0x20002919 0xdeadbeef
0x20002918 0xdeadbeef
0x20002917 0xdeadbeef
0x20002916 0xdeadbeef
0x20002915 0xdeadbeef
0x20002914 0xdeadbeef
0x20002913 0xdeadbeef
0x20002912 0xdeadbeef
0x20002911 0xdeadbeef
0x20002910 0xdeadbeef
0x2000290f 0xdeadbeef
0x2000290e 0xdeadbeef
0x2000290d 0xdeadbeef
0x2000290c 0xdeadbeef
0x2000290b 0xdeadbeef
0x2000290a 0xdeadbeef
0x20002909 0xdeadbeef
0x20002908 0xdeadbeef
0x20002907 0xdeadbeef
0x20002906 0xdeadbeef
0x20002905 0xdeadbeef
0x20002904 0xdeadbeef
0x20002903 0xdeadbeef
0x20002902 0xdeadbeef
0x20002901 0xdeadbeef
0x20002900 0xdeadbeef
0x200028ff 0xdeadbeef
0x200028fe 0xdeadbeef
0x200028fd 0xdeadbeef
0x200028fc 0xdeadbeef
0x200028fb 0xdeadbeef
0x200028fa 0xdeadbeef
0x200028f9 0xdeadbeef
0x200028f8 0xdeadbeef
0x200028f7 0x00000000
0x200028f6 0x00000000
0x200028f5 0x00000000
0x200028f4 0x00000000
0x200028f3 0x00000000
0x200028f2 0x00000000
0x200028f1 0x00000000
0x200028f0 0x00000000
0x200028ef 0x00000000
0x200028ee 0x10000000
0x200028ed 0xffffffff
0x200028ec 0x08000000
0x200028eb 0x00005300
0x200028ea 0xffffffff
0x200028e9 0x0814d800
0x200028e8 0x00000000
0x200028e7 0x00000000
0x200028e6 0x00000000
0x200028e5 0x00000000
0x200028e4 0x00000000
0x200028e3 0x00000000
0x200028e2 0x0802510f
0x200028e1 0x00000000
0x200028e0 0x20002b08
0x200028df 0x2000daf0
0x200028de 0x80000b20
0x200028dd 0x000003c0
0x200028dc 0x10004b90
0x200028db 0xdeadbeef
0x200028da 0x00000000
0x200028d9 0x00000000
0x200028d8 0x00000000
0x200028d7 0x00000000
0x200028d6 0x00000000
0x200028d5 0x00000000
0x200028d4 0x0801d93d
0x200028d3 0x200146c0
0x200028d2 0x20014530
User sp memory region, stack pointer lies within stack
0x20015049 0x10000000
0x20015048 0xffffffff
0x20015047 0x08000000
0x20015046 0x00005300
0x20015045 0xffffffff
0x20015044 0x0814d800
0x20015043 0x00000000
0x20015042 0x00000000
0x20015041 0x00000000
0x20015040 0x00000000
0x2001503f 0x00000000
0x2001503e 0x00000000
0x2001503d 0x0802510f
0x2001503c 0x00000000
0x2001503b 0x20002b08
0x2001503a 0x2000daf0
0x20015039 0x80000b20
0x20015038 0x000003c0
0x20015037 0x10004b90
0x20015036 0xdeadbeef
0x20015035 0x00000000
0x20015034 0x00000000
0x20015033 0x00000000
0x20015032 0x00000000
0x20015031 0x00000000
0x20015030 0x00000000
0x2001502f 0x0801d93d
0x2001502e 0x200146c0
0x2001502d 0x20014530
0x2001502c 0x08098c25
0x2001502b 0x08098c43
0x2001502a 0x00000000
0x20015029 0x00000000
0x20015028 0x00000000
0x20015027 0x00000000
0x20015026 0x20013024
0x20015025 0x20013000
0x20015024 0x2000daf0
0x20015023 0x20015220
0x20015022 0x00000000
0x20015021 0x00000000
0x20015020 0x00000000
0x2001501f 0x00000000
0x2001501e 0x0000ff97
0x2001501d 0x00000000
0x2001501c 0x00000000
0x2001501b 0x00000000
0x2001501a 0x00000000
0x20015019 0x00000000
0x20015018 0x00000000
0x20015017 0x00000000
0x20015016 0x00000000
0x20015015 0x00000000
0x20015014 0x00000000
0x20015013 0x00000000
0x20015012 0x00000000
0x20015011 0x00000000
0x20015010 0x00000000
0x2001500f 0x00000000
0x2001500e 0x0033d001
0x2001500d 0x1a0f0fcc
0x2001500c 0x71fe5ef1
0x2001500b 0x05021571
0x2001500a 0x3d26e5ec
0x20015009 0x011d0f0f
0x20015008 0x9de8f6e0
0x20015007 0x5c05bf31
0x20015006 0x72462c05
0x20015005 0xa5011f0f
0x20015004 0x0fbad9fb
0x20015003 0xfdb2052d
0x20015002 0xb65a3f3b
0x20015001 0xa062001b
0x20015000 0x0f0eafb8
0x20014fff 0x0e09a207
0x20014ffe 0x1644d043
0x20014ffd 0x6e32c800
0x20014ffc 0x150f0f97
0x20014ffb 0x5705c503
0x20014ffa 0x07a50849
0x20014ff9 0x48bc8b2f
0x20014ff8 0x001a0f0f
0x20014ff7 0xcb6efde8
0x20014ff6 0xf2066d57
0x20014ff5 0x263d26e5
0x20014ff4 0x5f001d0f
0x20014ff3 0x0fb382f4
0x20014ff2 0x4caf0760
0x20014ff1 0x07ec462c
0x20014ff0 0x051b000e
0x20014fef 0x0f0fb79b
0x20014fee 0xf3757307
0x20014fed 0x03a86d42
0x20014fec 0xbd1fdc00
0x20014feb 0x1f0f0fd7
0x20014fea 0xd9fad9db
0x20014fe9 0x06a554aa
0x20014fe8 0x3f3ba0c0
0x20014fe7 0x00170f0f
0x20014fe6 0xab40099c
0x20014fe5 0x38073301
0x20014fe4 0x0b447f98
0x20014fe3 0xc700100f
0x20014fe2 0x0fd21604
0x20014fe1 0x372c067c
0x20014fe0 0xd1c33dba
0x20014fdf 0x2bed2007
0x20014fde 0xc6000000
0x20014fdd 0x000bdace
0x20014fdc 0x28f90836
0x20014fdb 0x00009733
0x20014fda 0xff1a0000
0x20014fd9 0xcbfeff97
0x20014fd8 0x000000f0
0x20014fd7 0x00000000
0x20014fd6 0x000000f0
0x20014fd5 0x20015160
0x20014fd4 0x20014530
0x20014fd3 0x20014530
0x20014fd2 0x00000000
0x20014fd1 0x00000000
0x20014fd0 0x00000000
0x20014fcf 0x00000000
0x20014fce 0x00000000
0x20014fcd 0x00000000
0x20014fcc 0x00000000
0x20014fcb 0x00000000
0x20014fca 0x00000000
0x20014fc9 0x00000000
0x20014fc8 0x00000000
0x20014fc7 0x00000000
0x20014fc6 0x00000000
0x20014fc5 0x00000000
0x20014fc4 0x00000000
0x20014fc3 0x00000000
0x20014fc2 0x00000000
0x20014fc1 0xffffffe9
0x20014fc0 0x00000000
0x20014fbf 0x00000000
0x20014fbe 0x100055a0
0x20014fbd 0x00000000
0x20014fbc 0x00000000
0x20014fbb 0x000000f0
0x20014fba 0x200029f8
0x20014fb9 0x00000000
0x20014fb8 0x000000f0
0x20014fb7 0x200150e8
0x20014fb6 0x200156e8
0x20014fb5 0xc6083697
0x20014fb4 0x33ff1bcb
0x20014fb3 0xfeff9733
0x20014fb2 0xd0011a0f
0x20014fb1 0x0fcc71fe
0x20014fb0 0x5ef10502
0x20014faf 0x15713d26
0x20014fae 0x00000000
0x20014fad 0x00000101
0x20014fac 0x00000000
0x20014fab 0x00000003
0x20014faa 0x00000000
0x20014fa9 0x00000000
0x20014fa8 0x00000000
0x20014fa7 0x00000000
0x20014fa6 0x00000000
0x20014fa5 0x00000000
0x20014fa4 0x00000000
0x20014fa3 0x00000000
0x20014fa2 0x00000000
0x20014fa1 0x00000000
0x20014fa0 0x00000000
0x20014f9f 0x00003301
0x20014f9e 0x0b447f98
0x20014f9d 0xc700100f
0x20014f9c 0x0fd21604
0x20014f9b 0x372c067c
0x20014f9a 0xd1c33dba
0x20014f99 0x2bed2007
0x20014f98 0xc6000000<-- User sp
0x20014f97 0x000bdace
0x20014f96 0x00000010
0x20014f95 0x00000d90
0x20014f94 0x4559038e
0x20014f93 0x4d3ebc20
0x20014f92 0x00000000
0x20014f91 0x00000000
0x20014f90 0x00000000
0x20014f8f 0x00000000
0x20014f8e 0x00000000
0x20014f8d 0x00000000
0x20014f8c 0x00000000
0x20014f8b 0x00000000
0x20014f8a 0x00000000
0x20014f89 0x00000000
0x20014f88 0x00000000
0x20014f87 0x00000000
0x20014f86 0x00000000
0x20014f85 0x00000000
0x20014f84 0x000bdace
0x20014f83 0x0800b98f
0x20014f82 0x000000c8
0x20014f81 0x20003ab0
0x20014f80 0x10004c08
0x20014f7f 0x200145a8
0x20014f7e 0x100079a0
0x20014f7d 0x00000000
0x20014f7c 0x00000000
0x20014f7b 0x00000000
0x20014f7a 0x00000000
0x20014f79 0x00000000
0x20014f78 0x00000000
0x20014f77 0x00000000
0x20014f76 0x00000000
0x20014f75 0x00000000
0x20014f74 0x00000000
0x20014f73 0x00000000
0x20014f72 0x00000000
0x20014f71 0x00000000
0x20014f70 0x00000000
0x20014f6f 0x00000000
0x20014f6e 0x00000000
0x20014f6d 0xffffffe9
0x20014f6c 0x00000000
0x20014f6b 0x0000001c
0x20014f6a 0x00000000
0x20014f69 0x0813c1cc
0x20014f68 0x20015220
0x20014f67 0x28f9004a
0x20014f66 0x08360000
0x20014f65 0x00000000
0x20014f64 0x000000f0
0x20014f63 0x20014f98
0x20014f62 0x0000000b
0x20014f61 0xdace28f9
0x20014f60 0x004a0836
0x20014f5f 0x00000000
0x20014f5e 0x00000000
0x20014f5d 0x0033d001
0x20014f5c 0x1a0f0fcc
0x20014f5b 0x71fe5ef1
0x20014f5a 0x05021571
0x20014f59 0x3d26e5ec
0x20014f58 0x011d0f0f
0x20014f57 0x9de8f6e0
0x20014f56 0x5c05bf31
0x20014f55 0x72462c05
0x20014f54 0xa5011f0f
0x20014f53 0x0fbad9fb
0x20014f52 0xfdb2052d
0x20014f51 0xb65a3f3b
0x20014f50 0xa062001b
0x20014f4f 0x0f0eafb8
0x20014f4e 0x0e09a207
0x20014f4d 0x1644d043
0x20014f4c 0x6e32c800
0x20014f4b 0x150f0f97
0x20014f4a 0x5705c503
0x20014f49 0x07a50849
0x20014f48 0x48bc8b2f
0x20014f47 0x001a0f0f
0x20014f46 0xcb6efde8
0x20014f45 0xf2066d57
0x20014f44 0x263d26e5
0x20014f43 0x5f001d0f
0x20014f42 0x0fb382f4
0x20014f41 0x4caf0760
0x20014f40 0x07ec462c
0x20014f3f 0x051b000e
0x20014f3e 0x0f0fb79b
0x20014f3d 0xf3757307
0x20014f3c 0x03a86d42
0x20014f3b 0xbd1fdc00
0x20014f3a 0x1f0f0fd7
0x20014f39 0xd9fad9db
0x20014f38 0x06a554aa
0x20014f37 0x3f3ba0c0
0x20014f36 0x00170f0f
0x20014f35 0xab40099c
0x20014f34 0x38073301
0x20014f33 0x0b447f98
0x20014f32 0xc700100f
0x20014f31 0x0fd21604
0x20014f30 0x372c067c
0x20014f2f 0xd1c33dba
0x20014f2e 0x2bed2007
0x20014f2d 0xc6000000
0x20014f2c 0x000bdace
0x20014f2b 0x28f90836
0x20014f2a 0x20015288
0x20014f29 0x080969e9
0x20014f28 0x00000ad5
0x20014f27 0x00000001
0x20014f26 0x0000004b
0x20014f25 0x0800b005
0x20014f24 0x0813c1cc
0x20014f23 0x20015220
0x20014f22 0x2001509c
0x20014f21 0x20015220
0x20014f20 0x2000daf0
0x20014f1f 0x100079a0
0x20014f1e 0x000000cb
0x20014f1d 0xdeadbeef
0x20014f1c 0x0000004b
0x20014f1b 0x00000000
0x20014f1a 0x00000001
0x20014f19 0x0800b641
0x20014f18 0x20014eb0
0x20014f17 0x0000000f
0x20014f16 0x0814d864
0x20014f15 0x100079a0
0x20014f14 0x0814d864
0x20014f13 0x0800b969
0x20014f12 0x20014eb0
0x20014f11 0x0800b859
0x20014f10 0x0814d864
0x20014f0f 0x0800b8b9
0x20014f0e 0x0000004b
0x20014f0d 0x20014eb0
0x20014f0c 0x000000f0
0x20014f0b 0x100079a0
0x20014f0a 0x080250d5
0x20014f09 0x0802510f
0x20014f08 0x000000f0
0x20014f07 0x00000000
0x20014f06 0x100079a0
0x20014f05 0x08007513
0x20014f04 0x1000a2dc
0x20014f03 0x10004b90
0x20014f02 0x00000000
0x20014f01 0x08007373
0x20014f00 0x00000010
0x20014eff 0x00000d90
0x20014efe 0x4559038e
0x20014efd 0x4d3ebc20
0x20014efc 0x00000000
0x20014efb 0x00000000
0x20014efa 0x00000000
0x20014ef9 0x00000000
0x20014ef8 0x00000000
0x20014ef7 0x00000000
0x20014ef6 0x00000000
0x20014ef5 0x00000000
0x20014ef4 0x00000000
0x20014ef3 0x00000000
0x20014ef2 0x00000000
0x20014ef1 0x00000000
0x20014ef0 0x00000000
0x20014eef 0x21000000
0x20014eee 0x08011188
0x20014eed 0x08007373
0x20014eec 0x000000c8
0x20014eeb 0x10004b90
0x20014eea 0x10004c08
0x20014ee9 0x200145a8
0x20014ee8 0x00000002
0x20014ee7 0x00000000
0x20014ee6 0x00000000
0x20014ee5 0x00000000
0x20014ee4 0x00000000
0x20014ee3 0x00000000
[hardfault_log] -- 2000-01-01-01:46:31 END Fault Log --
```

**Bug Report** or **Feature Request**

For general questions please use [PX4 Discuss](http://discuss.px4.io/) or [Slack](http://slack.px4.io). Issue reports need to contain the items below:

* Download the flight log file from the vehicle ([tutorial](https://docs.px4.io/en/flight-reporting.html))
* Share the link to a log showing the problem on [PX4 Flight Review](http://logs.px4.io/)
* Expected behavior and actual behavior.
* Steps to reproduce the problem.
* Specifications like the version of the project, operating system, or hardware.

**Comment by bkueng** [comment]
Hi

Thanks for the report. Since you have additional code changes, it will be very hard to help you. Can you flash the officially released version (1.7.3) and see if you have the same issue?

Which hardware do you have attached to the board, and to which ports?

There's currently another hardfault report here: https://github.com/PX4/Firmware/issues/8913. It's different that the fault happens in `sem_post.c`, whereas here it's `armv7-m/up_hardfault.c`. But if it's due to memory corruption, it can still be the same cause.

**Comment by stmoon** [comment]
Hi, @bkueng 

yes, I modified something  to the firmware. However, I think that it is not related to the problem because it was working well until yesterday.

Maybe, I just replaced GPS. and it is connect to GPS2 (not GPS1). 
GPS1 connector is not used. (it just used for safety button).

I think it is related to the problem.. 

I have used pixhawk 2.1.
